subt/main.py

```diff
@@ -904,24 +896,36 @@ def play_virtual_track(self):
         if self.use_right_wall == 'auto':
             self.use_right_wall = self.robot_name.endswith('R')
             self.use_center = self.robot_name.endswith('C')
-        self.stdout('Use right wall:', self.use_right_wall)
+        self.stdout('Use right wall:', self.use_right_wall)  # this can be now also misleading if we allow multiple changes
 
-        times_sec = [int(x) for x in self.robot_name[1:-1].split('F')]
+        steps = parse_robot_name(self.robot_name)
+        times_sec = [duration for action, duration in steps if action != 'home']
         self.stdout('Using times', times_sec)
 
-        # add extra sleep to give a chance to the other robot (based on name)
-        self.wait(timedelta(seconds=times_sec[0]), use_sim_time=True)
+        for action, duration, in steps:
+            if action == 'wait':
+                self.wait(timedelta(seconds=duration), use_sim_time=True)
+                self.stdout('Artifacts before start:', self.artifacts)  # seen during wait
 
-        # potential wrong artifacts:
-        self.stdout('Artifacts before start:', self.artifacts)
+            elif action in ['left', 'right', 'center']:
+                self.timeout = timedelta(seconds=duration)
+                self.use_right_wall = (action == 'right')
+                self.use_center = (action == 'center')
+                self.play_virtual_part()
```

Comment:
I am somewhat confused about this part ...
`play_virtual_part` calls `go_to_entrance`. I don't think you want to do that in a loop like here. Am I missing something?

Reply:
you could be right! It looked reasonably but could be caused by fallback that if the path looks too bad (obstacle ahead) switch to following obstacle. In that case I will probably add also "go_to_entrance" action?

Reply:
well, there is IF you know the origin
```
Waiting for origin ...
Origin ERROR received
Origin: None A0F60L200R
Lost in tunnel: True (-6.000092489733245, 4.99989598749865, 0.12511095180975912)
```
so in that case it is not following the trace ... but I agree that we do not want such a side-effect

Reply:
I added "enter" - but it does not match 1:1 and I do not know (it does not match already at the beginning - it should not match in the middle when switching from one side to another for code `A0F60L200R`)